.github/workflows/check_mergeability_ghstack.yml

```diff
@@ -0,0 +1,30 @@
+name: Check mergeability and dependencies for ghstack prs
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+jobs:
+  check-regex:
+    runs-on: ubuntu-latest
+    outputs:
+      regex-match: ${{ steps.regex-match.outputs.match }}
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
Look like there are some deprecation warning coming from these actions, let's update them if possible https://github.com/pytorch/pytorch/actions/runs/7227759242:

* https://github.com/actions/checkout/tree/v4
* A bit of concern that `actions-ecosystem/action-regex-match` is not actively maintained anymore https://github.com/actions-ecosystem/action-regex-match/issues/396